addons/mail/wizard/mail_activity_schedule_wizard.py

```diff
@@ -0,0 +1,315 @@
+        res_model_ids = {r.res_model_id.id for r in self}
+        domain = expression.AND([
+            expression.OR([[('company_id', 'in', tuple(company_ids))], [('company_id', '=', False)]]),
+            expression.OR([[('res_model_ids', '=', False)],
+                           *[[('res_model_ids', '=', res_model_id)] for res_model_id in res_model_ids]])
+        ])
+        plan_ids_by_company_res_model = {
+            (company_id.id or False, res_model_id or False): cnt
+            for company_id, res_model_id, cnt in ActivityPlan._read_group(
+                domain, groupby=["company_id", "res_model_ids"], aggregates=["__count"])
+        }
+        for record in self:
+            record.has_plan = any(
+                plan_ids_by_company_res_model.get(company_res_model_id)
+                for company_res_model_id in ((False, False),
+                                             (False, record.res_model_id),
+                                             (record.company_id.id, False),
+                                             (record.company_id.id, record.res_model_id))
+            )
+
+    @api.depends('plan_id', 'res_model_id', 'res_ids',
+                 'on_demand_user_id',  # plan specific
+                 'activity_type_id', 'user_id')  # activity specific
+    def _compute_error(self):
+        for wizard in self:
+            errors = wizard._get_errors()
+
+            if errors:
+                if self.plan_id:
+                    error_display = _('The plan "%s" cannot be launched: <br><ul>', wizard.plan_id.name)
+                else:
+                    error_display = _('The activity cannot be launched: <br><ul>')
+                for error in errors:
+                    error_display += '<li>%s</li>' % error
+                error_display += '</ul>'
+            else:
+                error_display = False
+            wizard.error = error_display
+
+    @api.depends('error')
+    def _compute_has_error(self):
+        for wizard in self:
+            self.has_error = wizard.error and wizard.error != '<p><br></p>'
+
+    @api.depends('res_ids')
+    def _compute_is_batch_mode(self):
+        for wizard in self:
+            self.is_batch_mode = len(wizard._get_converted_res_ids(wizard.res_ids)) > 1
+
+    @api.onchange('activity_type_id')
+    def _onchange_activity_type_id(self):
+        self.env['mail.activity']._apply_activity_type_defaults(self.activity_type_id, self)
+
+    def _get_activities_to_schedule(self):
+        return self.plan_id.template_ids
+
+    @api.model
+    def _get_converted_res_ids(self, res_ids_str):
+        """ Convert list of ids separated by comma into list of id number. """
+        return [int(res_id_str) for res_id_str in res_ids_str.split(',')]
+
+    def _get_records(self):
+        return self.env[self.res_model].browse(self._get_converted_res_ids(self.res_ids))
+
+    @api.model
+    def _get_record_for_scheduling(self, record, responsible):
```

Comment:
Have to say this is a bit weird ... but don't have much ideas currently to avoid the override in HR. Maybe we could rename it ?
```suggestion
    def _redirect_activity_schedule(self, record, activity):
```